### ISSUE TITLE
Relative path schema loading

### ISSUE THREAD

**Issue by Vidminas** [redhat-developer/yaml-language-server#154]
With a custom schema configuration in the workspace, such as

```
"yaml.schemas" : {
      "./json_schema/my_schema.json" : "/*.yaml"
    }
```

The schema fails to load with the output message:

> [Error - 2:17:09 PM] Notification handler 'workspace/didChangeConfiguration' failed with message: [UriError]: Scheme is missing: {scheme: "", authority: "", path: "./json_schema/my_schema.json", query: "", fragment: ""}

This does not happen with the npm v0.4.1 yaml-language-server package, only with the latest GitHub build

**Comment by JPinkney** [comment]
Hmm, I'm guessing you created a *.vsix file with master of YAML language server and master of VSCode-YAML? Or are you using a different editor?

If yes, is [this commit](https://github.com/redhat-developer/vscode-yaml/commit/251a4ac7577dd18799c1804e1e150412371f4130) in the vsix?

**Comment by Vidminas** [comment]
That's pretty much it. Indeed, I have the latest commit in my working tree, so the commit you mention is included.

I've found that the error is caused by this call in `connection.onDidChangeConfiguration`:
```
jsonLanguageService.configure({
        schemas: jsonSchemas,
        validate: settings.yaml.validate,
        allowComments: true
    });
```
when jsonSchemas contains relative paths.

I think the trouble might have been introduced by [this commit](https://github.com/redhat-developer/yaml-language-server/commit/39e800c3ca4465cb71106759839064477eebefe7#diff-916156c481141bfcd08c6ae40ce10e00), specifically by the changes to `onDidChangeConfiguration`.

**Comment by Vidminas** [comment]
The last commit fixes the errors, but there might some underlying issues with autocomplete. For my YAML files, auto completion is not giving any suggestions (although hovering works).
No issues with the npm server version 🤔

**Comment by JPinkney** [comment]
Is it possible for you to post the schema? Its probably an issue inside of the autocompletor somewhere or when the schema is set. Autocompletion is the only feature we don't use https://www.npmjs.com/package/vscode-json-languageservice for

**Comment by Vidminas** [comment]
Looks like this issue was being caused by my schema. The relative path loading patch should be enough to fix the schema loading errors.

The fact that my schema was the culprit, revealed another issue - even though I had fixed the schema in the editor, the schema content was not reloaded after it was changed on disk. Only after closing and re-opening VS Code did the changes take effect.
I found that `registerExternalSchema` in `jsonSchemaService.ts` gets the URI: `file:///Users/vidminas/dev/yaml-langservice/test-workspace/json_schema/my_schema.json`
but when the schema is edited, `connection.onDidChangeWatchedFiles` in `server.ts` gets the URI: `file:///c%3A/Users/vidminas/dev/yaml-langservice/test-workspace/json_schema/my_schema.json`


This `c%3A` is the encoded prefix `C:`, so this issue would only affect Windows.
Changing deprecated `rootPath` to `rootUri` param fixes this inconsistency.

But this still was not enough, now schema changes were picked up, but content was still not reloaded.
Turns out that a call to `jsonLanguageService.resetSchema()` was missing.

**Comment by mawkler** [comment]
I'm getting (almost) the same error for **all YAML** files:

```
 "Request initialize failed with message: [UriError]: Scheme is missing: {scheme: \"\", authority: \"\", path: \"null\", query: \"\", fragment : \"\"}"
```

Here's a Neovim config to reproduce the issue. You launch it by doing `nvim -u path/to/file/below`:

<details>
<summary><b>Click to expand</b></summary>

```lua
local lazypath = vim.fn.stdpath("data") .. "/lazy/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
	vim.fn.system({
		"git",
		"clone",
		"--filter=blob:none",
		"https://github.com/folke/lazy.nvim.git",
		"--branch=stable",
		lazypath,
	})
end
vim.opt.rtp:prepend(lazypath)

require("lazy").setup({
	{
		"neovim/nvim-lspconfig",
		dependencies = {
			{ "williamboman/mason.nvim", config = true },
			"williamboman/mason-lspconfig.nvim",
			"WhoIsSethDaniel/mason-tool-installer.nvim",
		},
		config = function()
			require("mason").setup()
			require("mason-tool-installer").setup({
				ensure_installed = { "yamlls" },
			})

			require("mason-lspconfig").setup({
				handlers = {
					function(server_name)
						require("lspconfig")[server_name].setup({})
					end,
				},
			})
		end,
	},
})
```

</details>

I'm using yamlls version 1.15.0
